pkg/gadgets/run/types/metadata.go

```diff
@@ -155,3 +159,134 @@ func (m *GadgetMetadata) validateStructs(spec *ebpf.CollectionSpec) error {
 
 	return result
 }
+
+// Populate fills the metadata  from its ebpf spec
+func (m *GadgetMetadata) Populate(spec *ebpf.CollectionSpec) error {
+	if m.Tracers == nil {
+		m.Tracers = make(map[string]Tracer)
+	}
+
+	if m.Structs == nil {
+		m.Structs = make(map[string]Struct)
+	}
+
+	if err := m.populateTracers(spec); err != nil {
+		return fmt.Errorf("handling trace maps: %w", err)
+	}
+
+	return nil
+}
+
+func (m *GadgetMetadata) populateTracers(spec *ebpf.CollectionSpec) error {
+	traceMap := getTracerMapFromeBPF(spec)
+	if traceMap == nil {
+		log.Debug("No trace map found")
+		return nil
+	}
+
+	traceMapStruct, ok := traceMap.Value.(*btf.Struct)
+	if !ok {
+		return fmt.Errorf("BPF map %q does not have BTF info for values", traceMap.Name)
+	}
+
+	found := false
+
+	// TODO: this is weird but we need to check the map name as the tracer name can be
+	// different.
+	for _, t := range m.Tracers {
+		if t.MapName == traceMap.Name {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		log.Debugf("Adding tracer %q", traceMap.Name)
+		m.Tracers[traceMap.Name] = Tracer{
+			MapName:    traceMap.Name,
+			StructName: traceMapStruct.Name,
+		}
+	} else {
+		log.Debugf("Tracer using map %q already defined, skipping", traceMap.Name)
+	}
+
+	gadgetStruct := m.Structs[traceMapStruct.Name]
+	existingFields := make(map[string]struct{})
+	for _, field := range gadgetStruct.Fields {
+		existingFields[field.Name] = struct{}{}
+	}
+
+	for _, member := range traceMapStruct.Members {
+		// skip some specific members
+		if member.Name == "timestamp" {
+			log.Debug("Ignoring timestamp column: see https://github.com/inspektor-gadget/inspektor-gadget/issues/2000")
+			continue
+		}
+		// TODO: temporary disable mount ns as it'll be duplicated otherwise
+		if member.Type.TypeName() == gadgets.MntNsIdTypeName {
+			continue
+		}
+
+		// check if column already exists
+		if _, ok := existingFields[member.Name]; ok {
+			log.Debugf("Column %q already exists, skipping", member.Name)
+			continue
+		}
+
+		log.Debugf("Adding column %q", member.Name)
+		field := Field{
+			Name:        member.Name,
+			Description: "TODO: Fill field description",
+			Attributes: FieldAttributes{
+				Width:     16,
+				Alignment: AlignmentLeft,
+				Ellipsis:  EllipsisEnd,
+			},
```

Comment:
We should fill these things with the columns defaults for the types instead of hardcoding it here

Reply:
I'm not sure I get it. Would you mind rephrasing a bit please?

Reply:
If I understood the code correct every new field of a struct gets the following attributes:
```
Width:     16,
Alignment: AlignmentLeft,
Ellipsis:  EllipsisEnd,
```
This happens regardless of the type of the new field.
For example a bool field doesn't need `Width: 16`.

Reply:
I added some logic to handle it, would you mind checking it please?

Reply:
Looks good. I just saw that the `getWidthFromType` from the columns lib is private and a member function of `Columns[any]`.
In the future we could maybe use that. Ofc we still need the type conversion